### ISSUE TITLE
helloworld doesn't work [SOLVED]

### ISSUE THREAD

**Issue by petunja** [uprel/gisapp#23]
Hi and tank you for every suggestions.
I don't know how works github, so I apologize in case I would not write here, but I really don't know how to fix this problem.

I setup the client as clearly explained in the readme file.
When i try to open the helloworld example, I get a white page.
If I try to load helloworld without "?public=on" it gives me a login fail.
If I try to login I can't login with any user (also the guest user).
Everyway I try I can't load the map.

The only change I did from default setup operations is:
changed virtualhost on apache server, using port 81 for my old configuration of qgis-server.

when I try to connect to the page I have following errors in logs-files:
1) qgis-server-access.log:
`::1 - - [07/Aug/2016:19:53:06 +0200] "GET /gisapp/helloworld?public=on HTTP/1.1" 200 1019 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36"
::1 - - [07/Aug/2016:19:53:07 +0200] "POST /gisapp/admin/login.php HTTP/1.1"` **500** `343 "http://localhost:81/gisapp/helloworld?public=on" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/51.0.2704.79 Chrome/51.0.2704.79 Safari/537.36"`

2)qgis-server-error.log:
`[Sun Aug 07 19:53:07.321507 2016] [:error] [pid 4213] [client ::1:52071] PHP Fatal error:  Uncaught exception 'PDOException' with message 'could not find driver' in /var/www/html/gisapp/admin/class.Login.php:108\nStack trace:\n#0 /var/www/html/gisapp/admin/class.Login.php(108): PDO->__construct('pgsql:host=loca...', 'myuser', 'mypassword')\n#1 /var/www/html/gisapp/admin/class.Login.php(156): GisApp\\Login->createDatabaseConnection()\n#2 /var/www/html/gisapp/admin/login.php(38): GisApp\\Login->doLoginWithPostData()\n#3 {main}\n  thrown in /var/www/html/gisapp/admin/class.Login.php on line 108, referer: http://localhost:81/gisapp/helloworld?public=on`

can you help me?
tank you very much,
P.

**Comment by uprel** [comment]
Look your server error log, you are mising php-pgsql driver. Check PHP [requirements ](https://github.com/uprel/gisapp/wiki/1.-Requirements#4-php-55-works-also-with-php-7)here and install necessary.

**Comment by petunja** [comment]
fantastic! tank you.
I tried to install those requirements, but they seemed to not exixt.
I run on mint 17.3 with php 5.5.9 and the terminal installation aswered me:
`E: Unable to locate package php
E: Unable to locate package php-pgsql
E: Unable to locate package php-xml
E: Unable to locate package php-zip`

After your suggestion i looked on [Ubuntu official packages](http://packages.ubuntu.com/trusty/php/) list and I found: **php5** and **php5-pgsql**.

I'm a newbie so I don't understand every lib or requirement or package that I'm forced to use, but:
about **php-xml** and **php-zip** packages (also unabled) I found, on my php-info file that:
1) xml appears to be recognized and supported: 
   - libxml: Version	2.9.1 (active)
   - libXML streams:	enabled
   - SimpleXML: enabled
   - XML support: active
   - ... and so on
2) zip appears in field: Registered PHP Streams
I don't know if it's enough to work properly but I'm sure I will know it soon. :)

Tank you for your support, you did a great job.
I hope this post will help every dummy like me.
P.

p.s: I installed a lamp server with a simplified procedure: `$ sudo apt-get install lamp-server^`
so it could be a non appropriated way for installing all required dependencies.
